Modules/_uuidmodule.c

```diff
@@ -91,45 +121,1730 @@ py_windows_has_stable_node(void)
+    }
+    timestamp_ms = (uint64_t)(pytime / 1000000);
+
+    if (!state->last_timestamp_v7_init || timestamp_ms > state->last_timestamp_v7) {
+        if (uuid7_get_counter_and_tail(state, &counter, bytes + 12) < 0) {
+            return NULL;
+        }
+    } else {
+        if (timestamp_ms < state->last_timestamp_v7) {
+            timestamp_ms = state->last_timestamp_v7 + 1;
+        }
+        // advance the 42-bit counter
+        counter = state->last_counter_v7 + 1;
+        if (counter > 0x3FFFFFFFFFF) {
+            // advance the 48-bit timestamp
+            timestamp_ms += 1;
+            if (uuid7_get_counter_and_tail(state, &counter, bytes + 12) < 0) {
+                return NULL;
+            }
+        } else {
+            // This is the common fast path, we only need 4 bytes of entropy
+            // 32-bit random data
+            if (gen_random(state,  bytes + 12, 4) < 0) {
+                return NULL;
+            }
+        }
+    }
+
+    timestamp_ms &= 0xFFFFFFFFFFFF;
+    bytes[0] = (timestamp_ms >> 40);
+    bytes[1] = (timestamp_ms >> 32);
+    bytes[2] = (timestamp_ms >> 24);
+    bytes[3] = (timestamp_ms >> 16);
+    bytes[4] = (timestamp_ms >> 8);
+    bytes[5] = timestamp_ms;
+
+    uint16_t counter_hi = (counter >> 30) & 0x0FFF;
+    bytes[6] = 0x70 | ((counter_hi >> 8));  // Version 7 = 0111
+    bytes[7] = counter_hi;
+
+    uint16_t counter_mid = (counter >> 16) & 0x3FFF;
+    bytes[8] = 0x80 | (counter_mid >> 8);  // Variant = 10
+    bytes[9] = counter_mid;
+
+    uint16_t counter_lo = counter & 0xFFFF;
+    bytes[10] = counter_lo >> 8;
+    bytes[11] = counter_lo;
+
+    state->last_timestamp_v7_init = 1;
+    state->last_timestamp_v7 = timestamp_ms;
+    state->last_counter_v7 = counter;
+
+    return uuid_from_bytes_array(state->UuidType, bytes);
+}
+
+/*[clinic input]
+@critical_section
+_uuid._install_c_hooks
+
+    *
+    random_func: object
+    time_func: object
+
+[clinic start generated code]*/
+
+static PyObject *
+_uuid__install_c_hooks_impl(PyObject *module, PyObject *random_func,
+                            PyObject *time_func)
+/*[clinic end generated code: output=884aa6e91b2ea832 input=6c5017297067e2ea]*/
+{
+    uuid_state *state = get_uuid_state(module);
+
+    // Reset bufferred entropy -- tests need to always start fresh
+    // when the random function is changed. Reset last_timestamp_v7 --
+    // important for repeatable tests
+    state->random_idx = RANDOM_BUF_SIZE;
+    state->last_timestamp_v7_init = 0;
+
+    if (random_func == Py_None) {
+        Py_CLEAR(state->random_func);
+    } else {
+        Py_INCREF(random_func);
+        Py_XSETREF(state->random_func, random_func);
+    }
+
+    if (time_func == Py_None) {
+        Py_CLEAR(state->time_func);
+    } else {
+        Py_INCREF(time_func);
+        Py_XSETREF(state->time_func, time_func);
+    }
+
+    Py_RETURN_NONE;
+}
+
+/*[clinic input]
+_uuid.UUID.__init__
+
+    hex: 'U' = NULL
+    bytes: 'y*' = None
```

Comment:
```suggestion
    bytes: bytes = None
```

Using `'y*'` means accepting arbitrary bytes-like objects, but we only want to support bytes (see #73915). I'm not entirely sure about the other parameters for AC, but they should also accept explicitly passed `None`.